From 462bce790e6a7e68620a4ce260cc38f7ed0255d5 Mon Sep 17 00:00:00 2001
From: Swati Agarwal <swati.agarwal@xilinx.com>
Date: Wed, 17 Aug 2022 11:41:24 +0530
Subject: [PATCH] dmaengine: xilinx_dma: cleanup for fetching xlnx,num-fstores property
Git-commit: 462bce790e6a7e68620a4ce260cc38f7ed0255d5
Patch-mainline: v6.0-rc7
References: git-fixes

Free the allocated resources for missing xlnx,num-fstores property.

Signed-off-by: Swati Agarwal <swati.agarwal@xilinx.com>
Link: https://lore.kernel.org/r/20220817061125.4720-3-swati.agarwal@xilinx.com
Signed-off-by: Vinod Koul <vkoul@kernel.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/dma/xilinx/xilinx_dma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/xilinx/xilinx_dma.c b/drivers/dma/xilinx/xilinx_dma.c
index 97633f0b8114..5e6e46437aea 100644
--- a/drivers/dma/xilinx/xilinx_dma.c
+++ b/drivers/dma/xilinx/xilinx_dma.c
@@ -3071,7 +3071,7 @@ static int xilinx_dma_probe(struct platform_device *pdev)
 		if (err < 0) {
 			dev_err(xdev->dev,
 				"missing xlnx,num-fstores property\n");
-			return err;
+			goto disable_clks;
 		}
 
 		err = of_property_read_u32(node, "xlnx,flush-fsync",
-- 
2.35.3

